### ISSUE TITLE
Eval function from ExprEvaluator is building an expression not evaluation her

### ISSUE THREAD

**Issue by paulobuchsbaum** [axkr/symja_android_library#133]
### Is this a request for a feature or a bug report? 
Bug
### What is the current behavior? 
`Eval` from `ExprEvaluator`  don't evaluate the expression,  just  build her.
### If this is a bug report, please provide steps to reproduce: 

_MainActivity.kt_ written in _Kotlin_ (fully interoperable with _Java_)  inside _Android Studio 3.4.1_:
```
// Kotlin don't demand here a class wrapper for functions and don't  need 'new' token
// 'val' is read only variable, type is static and can be infered looking at right  expression. 
// 'val' is read/write variable
        var result=""
        val util = ExprEvaluator(false, 100)  // Kotlin don't demand here a class
        
        try {
            result = util.eval("diff(sin(x)*cos(x),x)").toString()
        } catch(e:Exception) {    // type in Kotlin is after the variable
           result = "error"
        }
```

Result: `D(sin(x)*cos(x),x)`

When I replace 
```
result = util.eval("diff(sin(x)*cos(x),x)").toString()
```
for
```
val expr  = util.parse("diff(sin(x)*cos(x),x)")
result =util.eval(expr).toString()
```

The  output is equal.

### What is the expected or desired behavior?  
  The derivative of above function in `Result`:  `Cos(x)^2-Sin(x)^2`
### Please provide use cases for changing the current behavior:
  Always
### Please describe your development environment (OS, browser, etc.): 
   _Android Studio 3.4.1_ with _Kotlin_

**Comment by tranleduy2000** [comment]
@paulobuchsbaum please checkout #129

**Comment by paulobuchsbaum** [comment]
Thanks for feedback,  @tranleduy2000!

I have not seen it. It is not clear for me in documentation that I've seen.  

So I've inserted `await` call before `ExprEvaluator` call and before first `eval` call. ( Android Studio inserts `import org.matheclipse.core.expression.F.await` automatically) 

So I run my demo  in an Android cell phone and in both cases the program was frozen

When I ran my demo with debugger, I noticed that the hangup is just in the `await ` call  line.

**Comment by tranleduy2000** [comment]
The `MainActivity.kt` should be written as below:
```Java
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        F.await()

        val util = ExprEvaluator(false, 100)  // Kotlin don't demand here a class

        try {
            val result: String = util.eval("diff(sin(x)*cos(x),x)").toString()
            println("result = $result")
        } catch (e: Exception) {    // type in Kotlin is after the variable
            e.printStackTrace()
        }
    }
}
``` 

Click run, see the Android Logcat:

![image](https://user-images.githubusercontent.com/19369448/59477957-c2b3cf00-8e81-11e9-8e1d-5b08fcac89eb.png)

Let's see how it works in Java
```Java
	/**
	 * Causes the current thread to wait until the main initialization has finished.
	 *
	 */
	public final static void await() throws InterruptedException {
		COUNT_DOWN_LATCH.await();
	}
```

The `COUNT_DOWN_LATCH` field is in static scope, so the static block will be called and all symbols will be initialized.

**Comment by paulobuchsbaum** [comment]
Thanks again  for your attention, @tranleduy2000 

My _Android Studio_ is completely updated and I use exactly the same code as you, but my code keeps on hanging in `await()` call, both in emulated device and in the real cell phone. 

It should be some project setting that  I don't get to imagine. 

I've tested without `instant run` setting, and it's the same thing.

PS: I've downloaded the JARs files in [Version_2019-05-23](https://github.com/axkr/symja_android_library/releases/tag/version_2019-05-23)   and put the JAR files in
`libs` folder. 

Axel told me that you have AAR version in [symja Android Library](https://github.com/tranleduy2000/symja_android_library/releases). Maybe I can try with put this library in `libs` folder in my local project.

**Comment by tranleduy2000** [comment]
Could you upload your `build.gradle` file and logcat before it was frozen?

**Comment by paulobuchsbaum** [comment]
Thanks a lot for your good will, @tranleduy2000 ! 

By the way, I also try all steps in [Gradle Usage](https://github.com/tranleduy2000/symja_android_library#gradle-usage) of your fork, but I've noticed that it does not change the `eval` and `exprEvaluator` calls. It continues linked to the old imports.  

```
import org.matheclipse.core.eval.ExprEvaluator
import org.matheclipse.core.expression.F
```

I erase all _logcat_ content and run the code with `F.await()` enabled, after I save all text in a text file. 
The _build.grade_ usages the .JAR files in the lib folder from [ Version_2019-05-23](https://github.com/axkr/symja_android_library/releases/tag/version_2019-05-23)  

[LogcatGradle.zip](https://github.com/axkr/symja_android_library/files/3288949/LogcatGradle.zip)

**Comment by tranleduy2000** [comment]
Oh! There is a big problem with `Apfloat` because it uses reflection API.

The following line in gradle file will cause enable proguard.
```Gradle
minifyEnabled true
```

So you can try to set `minifyEnabled` to false to avoid this problem.

**Comment by paulobuchsbaum** [comment]
Thanks, Duy Tran Le, but unfortunately it doesn't work.  It was a change in the past that I'd made to try solve the hanging problem. 

Other tries

a) I've used an updated version of `androidx.core:core` (1.10) without success.

b) I've disabled  `F.await` call and put `ExprEvaluator` call after I press a l button in layout. 
So I wait 10/20 seconds before calculation.  However, it keeps on parsing the expression without evaluate it.
